main.go

```diff
@@ -3,13 +3,38 @@ package main
 import (
 	"log"
 	"net/http"
+	"net/http/httputil"
+	"net/url"
 	"observability-remote-write-proxy/pkg/remotewrite"
 )
 
+const (
+	prefixHeader = "X-Forwarded-Prefix"
+)
+
 func main() {
-	handler := http.HandlerFunc(handleRequestSuccess)
-	http.Handle("/", handler)
-	err := http.ListenAndServe(":8080", nil)
+	upstreamUrl, err := url.Parse("http://localhost:9091/api/v1/write")
+	if err != nil {
+		panic(err)
+	}
+
+	proxy := httputil.ReverseProxy{
+		Director: func(request *http.Request) {
+			request.URL.Scheme = upstreamUrl.Scheme
+			request.Host = upstreamUrl.Host
+			request.URL.Host = upstreamUrl.Host
+			request.URL.Path = upstreamUrl.Path
+			request.Header.Add(prefixHeader, "/")
+		},
+		ErrorHandler: func(writer http.ResponseWriter, request *http.Request, err error) {
+			log.Println(err)
+		},
+	}
+
+	proxy.Transport = http.DefaultTransport
```

Comment:
We'll need to hook in the token refreshment source in this.
```
proxy.Transport = &oauth2.Transport{
		Source: ....Token refreshment source goes here,
		Base:   http.DefaultTransport,
	}
```

Reply:
/cc @pb82

Reply:
@machi1990 that will be done in a separate PR